std/algebra/emulated/sw_bn254/pairing.go

```diff
@@ -0,0 +1,283 @@
+package sw_bn254
+
+import (
+	"fmt"
+
+	"github.com/consensys/gnark/frontend"
+	"github.com/consensys/gnark/std/algebra/emulated/fields_bn254"
+	"github.com/consensys/gnark/std/math/emulated"
+)
+
+type Pairing struct {
+	*fields_bn254.Ext12
+}
+
+func NewPairing(api frontend.API) (*Pairing, error) {
+	ba, err := emulated.NewField[emulated.BN254Fp](api)
+	if err != nil {
+		return nil, fmt.Errorf("new base api: %w", err)
+	}
+	return &Pairing{
+		Ext12: fields_bn254.NewExt12(ba),
+	}, nil
+}
+
+func (pr Pairing) DoubleStep(p *g2Projective) (*g2Projective, *lineEvaluation) {
```

Comment:
Imo if not necessary to export, then I would make private. It would be easier for the users to find.

Reply:
This function will change anyway in the upcoming perf PR. Let's keep this refactoring PR minimal and will make the suggested change then.